src/Compilers/CSharp/Portable/Generated/CSharp.Generated.g4

```diff
@@ -663,7 +663,7 @@ anonymous_function_expression
   ;
 
 anonymous_method_expression
-  : 'async'? 'delegate' parameter_list? block expression?
+  : modifier* 'delegate' parameter_list? block expression?
```

Comment:
in terms of grammar generator, we could consider adding information to Syntax.xml stating what subset of modifiers are legal.  but no need to do that any time soon.